### ISSUE TITLE
KeyNotFoundException 

### ISSUE THREAD

**Issue by rozalba89** [stoneharry/WoW-Spell-Editor#105]
**Spell editor version**
Release Build v1.9.3, win10

**Describe the bug**
Loading Spell.dbc for 3.3.5 crashes with KeyNotFoundException 
My spell.dbc contains more than 60000 rows
(i tried with different sets of spells - never loaded more than 50000)

**To Reproduce**
1. Go to Import/export
2. Click on import selected files (custom spell.dbc wich contain more than 50000 rows)
3. Loading process starts normally and then stops with exception
4. in database i can see only 50000 rows

**Expected behavior**
Expect to load all the rows

**Screenshots**
![6b1a50313c4278f657442127c1db0cf5](https://user-images.githubusercontent.com/7187755/79114348-bba06a80-7d8b-11ea-8214-342102611cc2.png)

**Comment by stoneharry** [comment]
Can you upload your DBC files for me to test with please? I will not be able to reproduce this using the enUS Blizzlike ones.

I also noticed you are using ruRU client - make sure that all of the DBC files being used are for the ruRU client. Do not just add the ruRU spell.dbc while leaving the other DBC's default. See #55 for more details on this - the program does not support mixed locale dbc files.

**Comment by rozalba89** [comment]
Problem still present in Release Build v1.9.7

![66afdaac9fc1fea75ffe8f67fc676197](https://user-images.githubusercontent.com/7187755/89109298-84dc8600-d448-11ea-8ad0-f84db4e8bc20.png)
![a013863f90276373153bbf6ade88c32e](https://user-images.githubusercontent.com/7187755/89109299-85751c80-d448-11ea-933f-802086817c1b.png)

[My_DBC_wotlk.zip](https://github.com/stoneharry/WoW-Spell-Editor/files/5011139/My_DBC_wotlk.zip)

**Comment by stoneharry** [comment]
So this seems like a very corrupt DBC to me, but I managed to force it to load:

![screenshot](https://i.imgur.com/SXyYTcO.png)

Some of the string fields where pointing to string offsets that did not exist in the file. Instead of throwing an exception (which is what you were seeing) I have changed it to now return `Unknown String: <offset>`. The disadvantage of this approach is that it is silently failing - there is no easy way to see which strings have been modified. I can add logging to the `debug_output.txt` to highlight this.

There was errors in the spell visual DBCs where multiple rows existed with the same ID. The database enforces a primary key on the ID column, so it fails when importing.
